packages/formik/src/Formik.tsx

```diff
@@ -631,7 +631,7 @@ export function useFormik<Values extends FormikValues = FormikValues>({
       if (!isString(eventOrTextValue)) {
         // If we can, persist the event
         // @see https://reactjs.org/docs/events.html#event-pooling
-        if (!!(eventOrTextValue as React.ChangeEvent<any>).persist) {
+        if ((eventOrTextValue as any).persist) {
```

Comment:
This confuses me. eventOrTextValue is already `string | React.ChangeEvent<any>`, so if `!isString(eventOrTextValue)` then eventOrTextValue will automatically be inferred as React.ChangeEvent<any>, which means there shouldn't be a cast here at all. Are we using an old typescript version? Or the wrong definition of `isString` which does not use [type predicates](https://www.typescriptlang.org/docs/handbook/advanced-types.html#using-type-predicates)

```diff
-if ((eventOrTextValue as any).persist) {
+if (eventOrTextValue.persist) {
```

Reply:
Idk this didn’t have anything to do with this PR it just showed up in ts. reality is that the types might be lying (if we are in RN) and also that in react 17 this is a no op anyways.